proposals/native-integers.md

```diff
@@ -0,0 +1,342 @@
+But the set of operand types supported by C# is limited for simplicity and for consistency with existing operators in the language.
+
+Lifted versions of the operators, where the arguments and return types are `nint?` and `nuint?`, are supported.
+
+Compound assignment operations `x op= y` where `x` or `y` are native ints follow the same rules as with other primitive types with pre-defined operators.
+Specifically the expression is bound as `x = (T)(x op y)` where `T` is the type of `x` and where `x` is only evaluated once.
+
+The shift operators should mask the number of bits to shift appropriately
+(see [shift operators](https://github.com/dotnet/csharplang/blob/master/spec/expressions.md#shift-operators) in C# spec).
+
+### Dynamic
+
+The conversions and operators are synthesized by the compiler and are not part of the underlying `IntPtr` and `UIntPtr` types.
+As a result those conversions and operators _are not available_ from the runtime binder for `dynamic`. 
+
+```C#
+nint x = 2;
+nint y = x + x; // ok
+dynamic d = x;
+nint z = d + x; // RuntimeBinderException: '+' cannot be applied 'System.IntPtr' and 'System.IntPtr'
+```
+
+### Type members
+
+The only constructor for `nint` or `nuint` is the parameter-less constructor.
+
+The following members of `System.IntPtr` and `System.UIntPtr` _are explicitly excluded_ from `nint` or `nuint`:
+```C#
+// constructors
+// arithmetic operators
+// implicit and explicit conversions
+public static readonly IntPtr Zero; // use 0 instead
+public static int Size { get; }     // use sizeof() instead
+public static IntPtr Add(IntPtr pointer, int offset);
+public static IntPtr Subtract(IntPtr pointer, int offset);
+public int ToInt32();
+public long ToInt64();
+public void* ToPointer();
+```
+
+The remaining members of `System.IntPtr` and `System.UIntPtr` _are implicitly included_ in `nint` and `nuint`. For .NET Framework 4.7.2:
+```C#
+public override bool Equals(object obj);
+public override int GetHashCode();
+public override string ToString();
+public string ToString(string format);
+```
+
+Interfaces implemented by `System.IntPtr` and `System.UIntPtr` _are implicitly included_ in `nint` and `nuint`.
+For instance if `IntPtr` implements `ISerializable, IEquatable<IntPtr>, IComparable<IntPtr>`,
+then `nint` implements `ISerializable, IEquatable<nint>, IComparable<nint>`.
+
+### Overriding, hiding, and implementing
+
+`nint` and `System.IntPtr`, and `nuint` and `System.UIntPtr`, are considered equivalent for overriding, hiding, and implementing.
+
+Overloads cannot differ by `nint` and `System.IntPtr`, and `nuint` and `System.UIntPtr`, alone.
+Overrides and implementations may differ by `nint` and `System.IntPtr`, or `nuint` and `System.UIntPtr`, alone.
+Methods hide other methods that differ by `nint` and `System.IntPtr`, or `nuint` and `System.UIntPtr`, alone.
+
+### Miscellaneous
+
+`nint` and `nuint` expressions used as array indices are emitted without conversion.
```

Comment:
>emitted without conversion [](start = 57, length = 26)

It isn't clear what "emitted without conversion" means. The specification for array indexing requires a conversion, and the array indexing IL instructions have specific requirements for the type of the index.

Reply:
> array indexing IL instructions have specific requirements for the type of the index

IL instructions like `ldelem` and `ldelema` take either a `native int` or an `int32`